iconservice/deploy/icon_score_deployer.py

```diff
@@ -37,11 +39,9 @@ def deploy(self, address: Address, data: bytes, tx_hash: bytes):
         converted_tx_hash = f'0x{bytes.hex(tx_hash)}'
         install_path = os.path.join(score_root_path, converted_tx_hash)
         try:
-            if os.path.isfile(install_path):
-                raise ScoreInstallException(f'{install_path} is a file. Check your path.')
-            if os.path.isdir(install_path):
-                raise ScoreInstallException(f'{install_path} is a directory. Check {install_path}')
-            if not os.path.exists(install_path):
+            if os.path.exists(install_path):
```

Comment:
I think you ought to add mkdir logic previous that logic.
and you should add integrate-test about that